### DISCUSSION TITLE
Getting duplicate [ GET ] Request on 'getStaticProps' at Vercel.

### DISCUSSION THREAD

**Discussion by dev-xo** [vercel/vercel#6581] [Help] [answered]
Hello guys, I'm fetching an Issue, and not sure if im doing something wrong, or maybe its just a vercel bug, because on **localhost**, seems like all is going as it should go.

Basically Im getting a duplicate **_[ GET Request ]_** after requesting a **Page** on my site.
Im gonna leave you here a screenshot of vercel logs, and a screenshot of the actuall code.

![Screen Shot 2021-08-11 at 19 07 07](https://user-images.githubusercontent.com/82425492/129072702-d4fa6403-f339-4356-89e9-227f3c28a4df.png)

Getting duplicate GET requests, the timestamps are exactly the same as it can be seen in the image.

![Screen Shot 2021-08-11 at 19 37 17](https://user-images.githubusercontent.com/82425492/129076840-e0033d1f-e89f-4f84-aa9f-c30345ea5ab0.png)

It's a very simple app, im getting some data from the database, the **Page** should be on CDN after the first request, and on each request after that, im updating a '**views**' field in database for each call.

The thing is, the '**views**' field, updates 2 times, as Im getting a duplicate [ GET Request ] to the Page, and not sure why is happening that. Could be because im doing an extra call to the database, after the page is on CDN?

_**EDIT: I've removed the [ Update ] request from the code, and Im still getting a duplicate [ GET ] call on the Page.
Guess this could be something about vercel? Because as I said, on localhost all seems to work pretty good.**_

Hope Im explaining myself right, this is my first discussion, if you guys wanna help and need more info from my side, im here for it. 
Thanks!

**Comment by leerob** [comment, marked as answer]
You're seeing two requests in the Vercel logs - one for the JSON payload of the data from `getStaticProps` - and the other for the actual page you're rendering! This is expected 👍

**Reply by dev-xo** [reply]
Ow, thank you so much for the reply! Yeah, that seems to be the story on here.
Thanks you!

**Reply by O4epegb** [reply]
I am not quite sure that @leerob is right here, with all the respect.

See my issue https://github.com/vercel/vercel/discussions/6536

I am experiencing similar problems, and only on Vercel. On VPS or localhost (production mode) it works differently.

**Reply by dev-xo** [reply]
Yeah, thanks for noticing mate.. In my case, I moved my code to getServerSideProps instead of getStaticProps, had no choice because I was experiencing the duplicate request thing...

Anyways, thanks for noticing! If you find a good solution would be cool if you can share!
